tests/test-generators.py

```diff
@@ -19,7 +19,14 @@
             output += f"  {c[0]}\n"
             for cwa in sorted(c[1], key=lambda x: (x.opcode, x.vars)):
                 output += f"    {cwa.opcode}"
-                for a in cwa.vars:
+
+                # special case to omit an empty hint from CREATE_COIN, to preserve the
+                # output from the test cases not using a hint
+                var = list(cwa.vars)
+                if cwa.opcode == 51 and len(cwa.vars) == 3 and len(cwa.vars[2]) == 0:
+                    var.pop(2)
```

Comment:
without this special case, I would have to insert an extra space " " at the end of every `CREATE_COIN` expected-output line in the existing test cases. I think that would muddy the waters a bit and not make it as obvious that all existing test cases are unaffected. It does have the issue that there's no test case ensuring that we *do* get empty hints unconditionally though.

However, I'm hoping to change this API to, instead of returning a list of list of condition codes + args, returns a concrete type with all conditions having concrete types.